app/src/main/java/io/gnosis/safe/ui/safe/share/ShareSafeViewModel.kt

```diff
@@ -0,0 +1,65 @@
+package io.gnosis.safe.ui.safe.share
+
+import android.graphics.Bitmap
+import android.graphics.Color
+import androidx.annotation.ColorInt
+import androidx.lifecycle.viewModelScope
+import io.gnosis.data.models.Safe
+import io.gnosis.data.repositories.EnsRepository
+import io.gnosis.data.repositories.SafeRepository
+import io.gnosis.safe.ui.base.AppDispatchers
+import io.gnosis.safe.ui.base.BaseStateViewModel
+import io.gnosis.safe.ui.transaction.NoSafeSelected
+import kotlinx.coroutines.withContext
+import pm.gnosis.crypto.utils.asEthereumAddressChecksumString
+import pm.gnosis.svalinn.common.utils.QrCodeGenerator
+import timber.log.Timber
+import javax.inject.Inject
+
+class ShareSafeViewModel
+@Inject constructor(
+    private val safeRepository: SafeRepository,
+    private val ensRepository: EnsRepository,
+    private val qrCodeGenerator: QrCodeGenerator,
+    appDispatchers: AppDispatchers
+) : BaseStateViewModel<ShareSafeState>(appDispatchers) {
+
+    override fun initialState(): ShareSafeState = ShareSafeState()
+
+    fun load() {
+        safeLaunch {
+            safeRepository.getActiveSafe()?.let { activeSafe ->
+                val ensName = runCatching { ensRepository.reverseResolve(activeSafe.address) }
+                    .onFailure { Timber.e(it) }
+                    .getOrNull()
+                val qrCode = runCatching {
+                    qrCodeGenerator.generateQrCodeSync(
+                        activeSafe.address.asEthereumAddressChecksumString(),
+                        512,
```

Comment:
is there any reason to specify dimensions and background explicitly?

Reply:
Those are the default values of the method for the rxJava implementation, which suited our purposes. If you check `svalinn` out. For some reason the bitmap gets generated with some padding already, which I found a bit annoying. Maybe this could be changed, but then we'd need a new release for `svalinn` I suppose. This forces us to pull in rx too :/